app/models/user.rb

```diff
@@ -14,17 +14,14 @@ class User < ActiveRecord::Base
   # Makes getters and setters
   attr_accessor :pin
 
-  validates :school, :first_name, :last_name, :presence => true
-  validates :email, :uniqueness => true
-  # Validation only occurs when a user record is being
+  # Validation's for email and pin only occurs when a user record is being
   # created on sign up. Does not occur when updating
   # the record.
+  validates :school, :first_name, :last_name, :presence => true
   validates_format_of :email, with: /\@schools.nyc\.gov/, message: ' should end in @schools.nyc.gov', :on => :create
-  validates_format_of :first_name, :last_name, :with => /^[a-z]+$/i
+  validates_format_of :first_name, :last_name, :with => /\A[^0-9`!@#\$%\^&*+_=]+\z/
   validates_format_of :alt_email,:with => Devise::email_regexp, :allow_blank => true, :allow_nil => true
-  # Validation only occurs when a user record is being
-  # created on sign up. Does not occur when updating
-  # the record.
+  validates :alt_email, :uniqueness => true, :allow_blank => true, :allow_nil => true
```

Comment:
We can change this to `validates :alt_email, uniqueness: true, allow_blank: true, allow_nil: true`

Reply:
Please add tests for validation.